src/main/java/jenkins/plugins/gerrit/client/ssh/GerritClientSSH.java

```diff
@@ -0,0 +1,188 @@
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.jcraft.jsch.Session;
+import com.urswolfer.gerrit.client.rest.http.changes.ChangesParser;
+import com.urswolfer.gerrit.client.rest.http.changes.ReviewResultParser;
+import jenkins.plugins.gerrit.client.GerritClient;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.jgit.transport.RemoteSession;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.Charset;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class GerritClientSSH implements GerritClient {
+
+    private final int timeout;
+    private final Logger logger = LoggerFactory.getLogger(GerritClientSSH.class);
+    private RemoteSession delegate;
+    private Gson gson = new Gson();
+    private ChangesParser changesParser = new ChangesParser(gson);
+    private ReviewResultParser reviewResultParser = new ReviewResultParser(gson);
+
+    public GerritClientSSH(RemoteSession remoteSession, int timeout) {
+        this.delegate = remoteSession;
+        this.timeout = timeout;
+    }
+
+    @Override
+    public Changes changes() {
+        return new ChangesSSH();
+    }
+
+    @Override
+    public String getScheme() {
+        return "SSH";
+    }
+
+    private JsonElement request(String request) throws IOException {
+        return request(request, null);
+    }
+
+    private JsonElement request(String request, String stdin) throws IOException {
+        logger.debug("Initiating request '{}'", request);
+        Process process = delegate.exec(request, timeout);
+        if (stdin != null) {
+            if (!isAlive(process)) {
+                throw new IOException("Failed to initiate request '" + request + "'. Socket closed before we could send data");
+            }
+            logger.debug("Writing to stdin: '{}'", stdin);
+            IOUtils.write(stdin, process.getOutputStream(), Charset.defaultCharset());
+            process.getOutputStream().flush();
+            logger.debug("Finished writing to stdin. Closing channel");
+            process.getOutputStream().close();
+        }
+        boolean finished = false;
+        try {
+            finished = waitFor(process, timeout, TimeUnit.MILLISECONDS);
+        } catch (InterruptedException e) {
+            throw new IOException("Interrupted while waiting for SSH process to finish", e);
+        }
+        if (!finished) {
+            throw new IOException("Process did not finish in " + TimeUnit.MILLISECONDS.toSeconds(timeout) + " seconds");
+        }
+        if (isAlive(process)) { //Should hopefully not happen, but we'll destroy just in case.
+            process.destroy();
+        }
+        if (process.exitValue() != 0) {
+            String errorStream = IOUtils.toString(process.getErrorStream(), Charset.defaultCharset());

```

Comment:
using default charset is not best practice as you depend on remote settings and not local settings